# Mini Project 1 — Analysis Notebook

**Your name:**  Melanie Song

**Dataset:**  restaurants_cleaned.csv (cleaned from 380K_US_Restaurants.csv)

**Date:**  May 20, 2026

In [41]:
# Setup — run this cell first
# If any package is missing, it will install automatically
import subprocess, sys

def install(pkg):
    subprocess.check_call([sys.executable, "-m", "pip", "install", pkg, "-q"])

for pkg in ["pandas", "plotly", "kaleido", "nbformat"]:
    try:
        __import__(pkg)
    except ImportError:
        print(f"Installing {pkg}...")
        install(pkg)

import pandas as pd
import plotly.express as px

print("Setup complete.")

Setup complete.


---

## Section 1 — Overview

**Dataset:** This is a dataset of 380K Restaurants (Mostly USA-Based) from Kaggle: https://www.kaggle.com/datasets/kwxdata/380k-restaurants-mostly-usa-based/data. This dataset contains approximately 380,000 restaurant records, including information such as restaurant name, category (cuisine type), rating, and address/location. 

**Why this dataset:** *This project connects to my interest in human-centered design by exploring how cultural identity is reflected and surfaced through food data on digital platforms. Understanding patterns of representation and visibility can help inform how platforms categorize, recommend, and present culturally diverse experiences to users.*

**Three analytical questions:**

1. Which Asian cuisines (e.g., Chinese, Japanese, Thai, Vietnamese, etc.) are most and least represented among restaurants in the dataset?
2. How does the representation of Asian cuisines vary across different U.S. states (e.g., which states have higher or lower proportions of Asian cuisine restaurants)?
3. Do average ratings differ across Asian cuisine types, and are certain cuisines consistently rated higher or lower than others?


**What a practitioner would do with these findings:** Interested parties can access this to understand the distribution of Asian restaurants across the US. Food finders can also use this to locate some restaurants with good ratings. 

---

## Section 2 — Data Profile


In [80]:
# Load your dataset
# Replace 'your_dataset.csv' with your actual filename.
# The file should be in the same folder as this notebook.
# If you're loading from an API result, replace pd.read_csv() with the appropriate call.
#
# Example (app review dataset from class):
# df = pd.read_csv('app_reviews_demo.csv')

df = pd.read_csv('380K_US_Restaurants.csv')  # ← replace with your filename

print(df.shape)
df.head()

(380358, 13)


/var/folders/nz/k06zv2_91mx8l7l3ws72b6680000gn/T/ipykernel_88688/2514341641.py:9: DtypeWarning: Columns (3,9,10,11,12) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv('380K_US_Restaurants.csv')  # ← replace with your filename


,Title,Link,Category,Rating,Website,Phone,Address,Images,Categories,Geo_Coordinates,Time_Zone,Latitude,Longitude
0,Dairy Queen Grill & Chill,https://www.google.com/maps/place/Dairy+Queen+...,Fast food restaurant,3.8,http://www.fourteenfoods.com/?y_source=1_ODk5N...,+1256-496-0404,"Dairy Queen Grill & Chill, 3143 US-280, Alexan...",[{'thumbnail': 'https://lh5.googleusercontent....,"['Fast food restaurant', 'Ice cream shop']","{'latitude': 32.9338695, 'longitude': -85.9704...",America/Chicago,32.93387,-85.970419
1,Jake's Restaurant,https://www.google.com/maps/place/Jake%27s+Res...,American restaurant,4.4,http://jakesonbroad.com/,+1256-234-4300,"Jake's Restaurant, 16 Broad St, Alexander City...",[{'thumbnail': 'https://lh5.googleusercontent....,['American restaurant'],"{'latitude': 32.945406, 'longitude': -85.953806}",America/Chicago,32.945406,-85.953806
2,Carib Kitchen,https://www.google.com/maps/place/Carib+Kitche...,Caribbean restaurant,4.9,https://carib-kitchen.webnode.com/,+1256-392-4433,"Carib Kitchen, 68 Broad St, Alexander City, AL...",[{'thumbnail': 'https://lh5.googleusercontent....,['Caribbean restaurant'],"{'latitude': 32.9446173, 'longitude': -85.954932}",America/Chicago,32.944617,-85.954932
3,Cazadores Mexican Restaurant,https://www.google.com/maps/place/Cazadores+Me...,Mexican restaurant,4.5,NaN,+1256-392-3991,"Cazadores Mexican Restaurant, 910 Cherokee Rd,...",[{'thumbnail': 'https://lh5.googleusercontent....,['Mexican restaurant'],"{'latitude': 32.935643, 'longitude': -85.952365}",America/Chicago,32.935643,-85.952365
4,La Posada Mexican Grill,https://www.google.com/maps/place/La+Posada+Me...,Mexican restaurant,4.3,http://www.laposadamexicangrill.net/,+1256-329-3005,"La Posada Mexican Grill, 3714 US-280, Alexande...",[{'thumbnail': 'https://lh5.googleusercontent....,"['Mexican restaurant', 'Latin American restaur...","{'latitude': 32.926932, 'longitude': -85.9649187}",America/Chicago,32.926932,-85.964919


**`df.head()`:** The first rows show one listing per row (name, category, rating, contact, location), so each record is a single restaurant profile.


In [43]:
# Check column types and missing values
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 380358 entries, 0 to 380357
Data columns (total 13 columns):
 #   Column           Non-Null Count   Dtype 
---  ------           --------------   ----- 
 0   Title            380358 non-null  object
 1   Link             345209 non-null  object
 2   Category         380302 non-null  object
 3   Rating           337877 non-null  object
 4   Website          285435 non-null  object
 5   Phone            331499 non-null  object
 6   Address          380234 non-null  object
 7   Images           345209 non-null  object
 8   Categories       345153 non-null  object
 9   Geo_Coordinates  172604 non-null  object
 10  Time_Zone        172604 non-null  object
 11  Latitude         380358 non-null  object
 12  Longitude        380141 non-null  object
dtypes: object(13)
memory usage: 37.7+ MB


**`df.info()`:** There are 380,358 rows × 13 columns, mostly `object` dtypes, with large gaps in geo/time zone and noticeable missing ratings and contact fields.

Each row represents a restaurant listing with fields for:
- identity (Title, Category, Categories)
- quality/engagement (Rating)
- contact/presence (Website, Phone, Link, Images)
- location (Address, Latitude, Longitude, Geo_Coordinates, Time_Zone)

as columns. 

In [44]:
# Summary statistics for numeric columns
df.describe()

,Title,Link,Category,Rating,Website,Phone,Address,Images,Categories,Geo_Coordinates,Time_Zone,Latitude,Longitude
count,380358,345209,380302,337877.0,285435,331499,380234,345209,345153,172604,172604,380358.000000,380141.000000
unique,123770,156220,716,82.0,121313,149571,180187,335601,34826,154590,56,274279.000000,230191.000000
top,Subway,https://www.google.com/maps/place/Hawkers+Asia...,Restaurant,4.4,https://www.dominos.com/en/?utm_source=google&...,+1 800-786-1000,"Hawkers Asian Street Food, 1001 Park St, Jacks...",[],['Restaurant'],"{'latitude': 33.775089, 'longitude': -84.2805889}",America/New_York,30.326381,-129.942709
freq,4188,14,56966,37896.0,1124,172,14,1154,29194,12,61506,15.000000,21.000000


**`df.describe()`:** Numeric summaries show ratings clustered high (~4.3 median) and coordinate ranges that look plausible for U.S.-centric listings.


In [45]:
df.isnull().sum()

Title                   0
Link                35149
Category               56
Rating              42481
Website             94923
Phone               48859
Address               124
Images              35149
Categories          35205
Geo_Coordinates    207754
Time_Zone          207754
Latitude                0
Longitude             217
dtype: int64

**`df.isnull().sum()`:** Missing values are uneven—few missing categories, but many missing ratings and location/contact fields, which limits some analyses unless rows are filtered or imputed thoughtfully.


**Data profile notes:**  
- The dataset has 380,358 rows and 13 columns
- Each row is a different restaurant listing with fields for identity (Title, Category, Categories), quality (rating), contact and presence (website, phone, link, images), and location (address, latitutde, longitude, geo coordinates, time zone)
- Out of the columns, Category is one that we will be focusing on. This category has 716 unique labels, from specific cruisines to broad labels
- Not all of the data in every column is complete - about 54.6% are missing in geo_coordinates and time_zone, 25% don't have websites, 12.8% don't have phone numbers, and 11.2% don't have ratings
- Ratings is originally loaded as text that is then converted to numeric. The mean rating is about 4.27, while the median is 4.3
- There are only 56 rows missing categories. However, 86,716 restaurants have a very broad category (restaurant), which hides cuisine details
- There may be some repeat entries, since duplicate Title + Address combinations are high (52.5%)
- About 61,412 rows match Asian-related category keywords
- Largest groups include Chinese, Thai, Japanese, Indian, Vietnamese



Data Cleaning and grouping

First, I want to check if there are duplicate entires by checking for combos of the same Title and Address

In [46]:
# check duplicate entries by combo of Title and Address
dup_rows = df[["Title", "Address"]].fillna("").duplicated().sum()
dup_pct = dup_rows / len(df) * 100

print("Total rows:", len(df))
print("Duplicate Title+Address rows:", dup_rows)
print(f"Percent duplicate: {dup_pct:.2f}%")

Total rows: 380358
Duplicate Title+Address rows: 199686
Percent duplicate: 52.50%


In [47]:
#visualize combos of Title and Address + count
(
    df.assign(Title=df["Title"].fillna(""), Address=df["Address"].fillna(""))
      .groupby(["Title", "Address"], dropna=False)
      .size()
      .reset_index(name="count")
      .sort_values("count", ascending=False)
      .head(10)
)

,Title,Address,count
64451,Hawkers Asian Street Food,"Hawkers Asian Street Food, 1001 Park St, Jacks...",14
153372,Tasty Pot,"Tasty Pot, 9802 Baymeadows Rd, Jacksonville, F...",12
105642,Newport Tokyo House,"Newport Tokyo House, 6 Equality Park Pl, Newpo...",12
16206,Blue Orchid Thai Cuisine,"Blue Orchid Thai Cuisine, 13475 Atlantic Blvd ...",12
56895,Gangnam Korean Restaurant,"Gangnam Korean Restaurant, 5161 Beach Blvd #5,...",12
109483,Olive Garden Italian Restaurant,"Olive Garden Italian Restaurant, 4860 N Hamilt...",10
106894,Noodles and Company,"Noodles and Company, 2560 E 3rd St, Bloomingto...",10
44783,Drift Cafe,"Drift Cafe, 190 Thames St, Newport, RI 02840, ...",10
133959,Salt Life Food Shack,"Salt Life Food Shack, 1018 3rd St N, Jacksonvi...",10
71302,Izakaya Japanese Restaurant & Sake Bar,"Izakaya Japanese Restaurant & Sake Bar, 39 Col...",10


The analysis before has shown that there are duplicate entires. To clean the data, I took some steps:

In [50]:
# 1) Build a clean duplicate key
work = df.copy()
for c in ["Title", "Address", "Phone", "Latitude", "Longitude", "Geo_Coordinates"]:
    if c in work.columns:
        work[c] = work[c].fillna("").astype(str).str.strip()

# Keep only rows where Title+Address appears more than once
dup_mask = work.duplicated(subset=["Title", "Address"], keep=False)
dups = work.loc[dup_mask].copy()

print("Duplicate rows (Title+Address):", len(dups))
print("Unique duplicate groups:", dups.groupby(["Title", "Address"]).ngroups)

Duplicate rows (Title+Address): 367492
Unique duplicate groups: 167806


In [51]:
# 2) For each duplicate group, count unique values in each field
group_check = (
    dups.groupby(["Title", "Address"], dropna=False)
        .agg(
            rows=("Title", "size"),
            phone_unique=("Phone", "nunique"),
            lat_unique=("Latitude", "nunique"),
            lon_unique=("Longitude", "nunique"),
            geo_unique=("Geo_Coordinates", "nunique"),
        )
        .reset_index()
)

# Flags: True means values differ within the same Title+Address group
group_check["phone_mismatch"] = group_check["phone_unique"] > 1
group_check["lat_mismatch"] = group_check["lat_unique"] > 1
group_check["lon_mismatch"] = group_check["lon_unique"] > 1
group_check["geo_mismatch"] = group_check["geo_unique"] > 1

# Summary
summary = {
    "groups_total": len(group_check),
    "phone_mismatch_groups": int(group_check["phone_mismatch"].sum()),
    "lat_mismatch_groups": int(group_check["lat_mismatch"].sum()),
    "lon_mismatch_groups": int(group_check["lon_mismatch"].sum()),
    "geo_mismatch_groups": int(group_check["geo_mismatch"].sum()),
}
summary

{'groups_total': 167806,
 'phone_mismatch_groups': 2192,
 'lat_mismatch_groups': 218,
 'lon_mismatch_groups': 219,
 'geo_mismatch_groups': 149807}

In [52]:
# 3) Show examples where something differs
examples = group_check[
    group_check[["phone_mismatch", "lat_mismatch", "lon_mismatch", "geo_mismatch"]].any(axis=1)
].sort_values("rows", ascending=False).head(10)

examples

,Title,Address,rows,phone_unique,lat_unique,lon_unique,geo_unique,phone_mismatch,lat_mismatch,lon_mismatch,geo_mismatch
60118,Hawkers Asian Street Food,"Hawkers Asian Street Food, 1001 Park St, Jacks...",14,1,1,1,2,False,False,False,True
98239,Newport Tokyo House,"Newport Tokyo House, 6 Equality Park Pl, Newpo...",12,1,1,1,2,False,False,False,True
142474,Tasty Pot,"Tasty Pot, 9802 Baymeadows Rd, Jacksonville, F...",12,1,1,1,2,False,False,False,True
15094,Blue Orchid Thai Cuisine,"Blue Orchid Thai Cuisine, 13475 Atlantic Blvd ...",12,1,1,1,2,False,False,False,True
52975,Gangnam Korean Restaurant,"Gangnam Korean Restaurant, 5161 Beach Blvd #5,...",12,1,1,1,2,False,False,False,True
45939,Eleven South,"Eleven South, 216 11th Ave S, Jacksonville Bea...",10,1,1,1,2,False,False,False,True
57343,Grill House,"Grill House, Food Court, 21182 Salmon Run Mall...",10,1,1,1,2,False,False,False,True
83784,Lucky Star Restaurant,"Lucky Star Restaurant, 1140 Meadow Ln, Concord...",10,1,1,1,2,False,False,False,True
101820,Olive Garden Italian Restaurant,"Olive Garden Italian Restaurant, 4860 N Hamilt...",10,1,1,1,2,False,False,False,True
52222,Fujiyama Japanese Restaurant,"Fujiyama Japanese Restaurant, 21 Regent St, Cl...",10,1,1,1,2,False,False,False,True


In [53]:
# 4) Drill into one mismatched group to inspect exact values
# (run after viewing examples)
from IPython.display import display

print("examples rows:", len(examples))

if len(examples) == 0:
    print("No mismatched duplicate groups found.")
else:
    t = examples.iloc[0]["Title"]
    a = examples.iloc[0]["Address"]
    print("Inspecting group:")
    print("Title:", t)
    print("Address:", a)

    out = dups[(dups["Title"] == t) & (dups["Address"] == a)][
        ["Title", "Address", "Phone", "Latitude", "Longitude", "Geo_Coordinates", "Link", "Categories"]
    ].head(20)

    print("Rows shown:", len(out))
    display(out)

examples rows: 10
Inspecting group:
Title: Hawkers Asian Street Food
Address: Hawkers Asian Street Food, 1001 Park St, Jacksonville, FL 32204, United States
Rows shown: 14


,Title,Address,Phone,Latitude,Longitude,Geo_Coordinates,Link,Categories
7171,Hawkers Asian Street Food,"Hawkers Asian Street Food, 1001 Park St, Jacks...",+1 904-508-0342,30.315006,-81.680353,"{'latitude': 30.315006, 'longitude': -81.680353}",https://www.google.com/maps/place/Hawkers+Asia...,['Asian restaurant']
34533,Hawkers Asian Street Food,"Hawkers Asian Street Food, 1001 Park St, Jacks...",+1 904-508-0342,30.315006,-81.680353,"{'latitude': 30.315006, 'longitude': -81.680353}",https://www.google.com/maps/place/Hawkers+Asia...,['Asian restaurant']
45401,Hawkers Asian Street Food,"Hawkers Asian Street Food, 1001 Park St, Jacks...",+1 904-508-0342,30.315006,-81.680353,"{'latitude': 30.315006, 'longitude': -81.680353}",https://www.google.com/maps/place/Hawkers+Asia...,['Asian restaurant']
109804,Hawkers Asian Street Food,"Hawkers Asian Street Food, 1001 Park St, Jacks...",+1 904-508-0342,30.315006,-81.680353,"{'latitude': 30.315006, 'longitude': -81.680353}",https://www.google.com/maps/place/Hawkers+Asia...,['Asian restaurant']
116979,Hawkers Asian Street Food,"Hawkers Asian Street Food, 1001 Park St, Jacks...",+1 904-508-0342,30.315006,-81.680353,"{'latitude': 30.315006, 'longitude': -81.680353}",https://www.google.com/maps/place/Hawkers+Asia...,['Asian restaurant']
160200,Hawkers Asian Street Food,"Hawkers Asian Street Food, 1001 Park St, Jacks...",+1 904-508-0342,30.315006,-81.680353,"{'latitude': 30.315006, 'longitude': -81.680353}",https://www.google.com/maps/place/Hawkers+Asia...,['Asian restaurant']
183600,Hawkers Asian Street Food,"Hawkers Asian Street Food, 1001 Park St, Jacks...",+1 904-508-0342,30.315006,-81.680353,"{'latitude': 30.315006, 'longitude': -81.680353}",https://www.google.com/maps/place/Hawkers+Asia...,['Asian restaurant']
252230,Hawkers Asian Street Food,"Hawkers Asian Street Food, 1001 Park St, Jacks...",+1 904-508-0342,30.315006,-81.680353,,https://www.google.com/maps/place/Hawkers+Asia...,['Asian restaurant']
314495,Hawkers Asian Street Food,"Hawkers Asian Street Food, 1001 Park St, Jacks...",+1 904-508-0342,30.315006,-81.680353,,https://www.google.com/maps/place/Hawkers+Asia...,['Asian restaurant']
350767,Hawkers Asian Street Food,"Hawkers Asian Street Food, 1001 Park St, Jacks...",+1 904-508-0342,30.315006,-81.680353,,https://www.google.com/maps/place/Hawkers+Asia...,['Asian restaurant']


In [54]:
subset = dups[(dups["Title"] == t) & (dups["Address"] == a)][
    ["Phone", "Latitude", "Longitude", "Geo_Coordinates", "Link", "Categories"]
]

print("Unique Phone:", subset["Phone"].nunique())
print("Unique Lat:", subset["Latitude"].nunique())
print("Unique Lon:", subset["Longitude"].nunique())
print("Unique Geo:", subset["Geo_Coordinates"].nunique())

subset.drop_duplicates()

Unique Phone: 1
Unique Lat: 1
Unique Lon: 1
Unique Geo: 2


,Phone,Latitude,Longitude,Geo_Coordinates,Link,Categories
7171,+1 904-508-0342,30.315006,-81.680353,"{'latitude': 30.315006, 'longitude': -81.680353}",https://www.google.com/maps/place/Hawkers+Asia...,['Asian restaurant']
252230,+1 904-508-0342,30.315006,-81.680353,,https://www.google.com/maps/place/Hawkers+Asia...,['Asian restaurant']


After cleaning data, there are 183,727 rows left. 

In [55]:
import pandas as pd

# Load cleaned dataset
df = pd.read_csv("restaurants_cleaned.csv", low_memory=False)

# Define Asian-related keyword pattern (word-boundary to avoid matching "caucasian")
pattern = r"\b(?:asian|chinese|japanese|thai|vietnamese|korean|indian|filipino|malaysian|indonesian|taiwanese)\b"

# Build filtered dataframe
asian_df = df[
    df["Category"].fillna("").str.lower().str.contains(pattern, regex=True)
].copy()

# Explicitly remove "Caucasian restaurant" just in case
asian_df = asian_df[
    asian_df["Category"].fillna("").str.lower() != "caucasian restaurant"
]

# Summary stats
total_rows = len(df)
asian_rows = len(asian_df)
asian_pct = (asian_rows / total_rows) * 100
num_categories = asian_df["Category"].nunique(dropna=True)

print("Total rows:", total_rows)
print("Asian-related rows (excluding Caucasian restaurant):", asian_rows)
print(f"Percent Asian-related: {asian_pct:.2f}%")
print("Unique Asian-related categories:", num_categories)

# Show categories counted
categories_used = sorted(asian_df["Category"].dropna().astype(str).str.strip().unique())
print("\nCategories included:")
for c in categories_used:
    print("-", c)

# Optional: top categories by count
print("\nTop 20 Asian-related categories:")
print(asian_df["Category"].value_counts().head(20))

Total rows: 183727
Asian-related rows (excluding Caucasian restaurant): 27195
Percent Asian-related: 14.80%
Unique Asian-related categories: 39

Categories included:
- Asian fusion restaurant
- Asian grocery store
- Asian restaurant
- Authentic Japanese restaurant
- Chinese bakery
- Chinese noodle restaurant
- Chinese restaurant
- Chinese supermarket
- Chinese takeaway
- Chinese tea house
- Delivery Chinese restaurant
- Filipino restaurant
- Indian Muslim restaurant
- Indian grocery store
- Indian restaurant
- Indian takeaway
- Indonesian restaurant
- Japanese curry restaurant
- Japanese delicatessen
- Japanese grocery store
- Japanese regional restaurant
- Japanese restaurant
- Japanese steakhouse
- Korean barbecue restaurant
- Korean grocery store
- Korean restaurant
- Korean rib restaurant
- Kyoto style Japanese restaurant
- Malaysian restaurant
- Modern Indian restaurant
- North Eastern Indian restaurant
- North Indian restaurant
- Pan-Asian restaurant
- South Asian restaurant
- So

In [56]:
df.head()


,Title,Link,Category,Rating,Website,Phone,Address,Images,Categories,Geo_Coordinates,Time_Zone,Latitude,Longitude
0,Dairy Queen Grill & Chill,https://www.google.com/maps/place/Dairy+Queen+...,Fast food restaurant,3.8,http://www.fourteenfoods.com/?y_source=1_ODk5N...,+1256-496-0404,"Dairy Queen Grill & Chill, 3143 US-280, Alexan...",[{'thumbnail': 'https://lh5.googleusercontent....,"['Fast food restaurant', 'Ice cream shop']","{'latitude': 32.9338695, 'longitude': -85.9704...",America/Chicago,32.9338695,-85.9704191
1,Jake's Restaurant,https://www.google.com/maps/place/Jake%27s+Res...,American restaurant,4.4,http://jakesonbroad.com/,+1256-234-4300,"Jake's Restaurant, 16 Broad St, Alexander City...",[{'thumbnail': 'https://lh5.googleusercontent....,['American restaurant'],"{'latitude': 32.945406, 'longitude': -85.953806}",America/Chicago,32.945406,-85.953806
2,Carib Kitchen,https://www.google.com/maps/place/Carib+Kitche...,Caribbean restaurant,4.9,https://carib-kitchen.webnode.com/,+1256-392-4433,"Carib Kitchen, 68 Broad St, Alexander City, AL...",[{'thumbnail': 'https://lh5.googleusercontent....,['Caribbean restaurant'],"{'latitude': 32.9446173, 'longitude': -85.954932}",America/Chicago,32.9446173,-85.954932
3,Cazadores Mexican Restaurant,https://www.google.com/maps/place/Cazadores+Me...,Mexican restaurant,4.5,NaN,+1256-392-3991,"Cazadores Mexican Restaurant, 910 Cherokee Rd,...",[{'thumbnail': 'https://lh5.googleusercontent....,['Mexican restaurant'],"{'latitude': 32.935643, 'longitude': -85.952365}",America/Chicago,32.935643,-85.952365
4,La Posada Mexican Grill,https://www.google.com/maps/place/La+Posada+Me...,Mexican restaurant,4.3,http://www.laposadamexicangrill.net/,+1256-329-3005,"La Posada Mexican Grill, 3714 US-280, Alexande...",[{'thumbnail': 'https://lh5.googleusercontent....,"['Mexican restaurant', 'Latin American restaur...","{'latitude': 32.926932, 'longitude': -85.9649187}",America/Chicago,32.926932,-85.9649187


The first rows still look like one restaurant per record (name, Category, Rating, contact fields, address/coordinates), but after cleaning there are far fewer repeated listings than in the raw export, so each row is closer to a single venue profile for analysis.

In [57]:
df.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 183727 entries, 0 to 183726
Data columns (total 13 columns):
 #   Column           Non-Null Count   Dtype 
---  ------           --------------   ----- 
 0   Title            183727 non-null  object
 1   Link             165453 non-null  object
 2   Category         183699 non-null  object
 3   Rating           161588 non-null  object
 4   Website          135099 non-null  object
 5   Phone            158404 non-null  object
 6   Address          183660 non-null  object
 7   Images           165453 non-null  object
 8   Categories       165425 non-null  object
 9   Geo_Coordinates  156996 non-null  object
 10  Time_Zone        156996 non-null  object
 11  Latitude         183727 non-null  object
 12  Longitude        183510 non-null  object
dtypes: object(13)
memory usage: 18.2+ MB


The cleaned table keeps the same 13 columns and 183,727 rows (down from ~380K), with most fields still stored as object text; core identity fields like Title, Address, and Latitude are nearly complete, while optional scrape fields (Link, Images, Website) still have noticeable gaps.


In [58]:
df.describe()


,Title,Link,Category,Rating,Website,Phone,Address,Images,Categories,Geo_Coordinates,Time_Zone,Latitude,Longitude
count,183727,165453,183699,161588,135099,158404,183660,165453,165425,156996,156996,183727,183510
unique,123770,156208,714,47,121236,149571,180187,163831,34704,154590,56,168096,167803
top,Subway,https://www.google.com/maps/place/Osaka+cullma...,Restaurant,4.4,https://www.dominos.com/en/?utm_source=google&...,4.41143E+11,"1580 S Melrose Dr #106, Vista",[],['Restaurant'],"{'latitude': 46.423669, 'longitude': -129.9427...",America/New_York,33.775089,-84.2805889
freq,2085,4,28074,21609,545,85,4,622,14791,11,55948,15,15


Numeric summaries on Rating (once coerced) show ratings clustered high—mean about 4.27, median 4.3, with most values between roughly 4.1 and 4.5—and latitude/longitude ranges that still look plausible for U.S.-centric listings, so the cleaned file is reasonable for comparing cuisine groups on stars and location.

In [59]:
df.isnull().sum()


Title                  0
Link               18274
Category              28
Rating             22139
Website            48628
Phone              25323
Address               67
Images             18274
Categories         18302
Geo_Coordinates    26731
Time_Zone          26731
Latitude               0
Longitude            217
dtype: int64

Missingness is lighter than in the raw file but still uneven: only 28 rows lack Category, about 12% lack Rating, roughly 14% lack Geo_Coordinates / Time_Zone, and Website / Phone are missing on a sizable share of rows—so Q3 (ratings) and Q2 (state from Address) are workable with filters, while geo-based work would drop or bias rows that lack coordinates.



Overall, in the cleaned dataset, there are 27,196 rows, with 40 unique categories that are Asian related. 

To address the missing ratings data, I filtered through the newly cleaned dataset again.

In [61]:
import pandas as pd

df = pd.read_csv("restaurants_cleaned.csv", low_memory=False)
#Rows with blank, non-numeric, or unparseable Rating values are removed (errors="coerce" turns those into NaN).

before = len(df)
df["rating_num"] = pd.to_numeric(df["Rating"], errors="coerce")
df = df[df["rating_num"].notna()].drop(columns=["rating_num"])
after = len(df)

df.to_csv("restaurants_cleaned.csv", index=False)

print(f"Rows before: {before:,}")
print(f"Rows after:  {after:,}")
print(f"Dropped:     {before - after:,}")

Rows before: 161,587
Rows after:  161,587
Dropped:     0


In [62]:
df.head()


,Title,Link,Category,Rating,Website,Phone,Address,Images,Categories,Geo_Coordinates,Time_Zone,Latitude,Longitude
0,Dairy Queen Grill & Chill,https://www.google.com/maps/place/Dairy+Queen+...,Fast food restaurant,3.8,http://www.fourteenfoods.com/?y_source=1_ODk5N...,+1256-496-0404,"Dairy Queen Grill & Chill, 3143 US-280, Alexan...",[{'thumbnail': 'https://lh5.googleusercontent....,"['Fast food restaurant', 'Ice cream shop']","{'latitude': 32.9338695, 'longitude': -85.9704...",America/Chicago,32.933870,-85.970419
1,Jake's Restaurant,https://www.google.com/maps/place/Jake%27s+Res...,American restaurant,4.4,http://jakesonbroad.com/,+1256-234-4300,"Jake's Restaurant, 16 Broad St, Alexander City...",[{'thumbnail': 'https://lh5.googleusercontent....,['American restaurant'],"{'latitude': 32.945406, 'longitude': -85.953806}",America/Chicago,32.945406,-85.953806
2,Carib Kitchen,https://www.google.com/maps/place/Carib+Kitche...,Caribbean restaurant,4.9,https://carib-kitchen.webnode.com/,+1256-392-4433,"Carib Kitchen, 68 Broad St, Alexander City, AL...",[{'thumbnail': 'https://lh5.googleusercontent....,['Caribbean restaurant'],"{'latitude': 32.9446173, 'longitude': -85.954932}",America/Chicago,32.944617,-85.954932
3,Cazadores Mexican Restaurant,https://www.google.com/maps/place/Cazadores+Me...,Mexican restaurant,4.5,NaN,+1256-392-3991,"Cazadores Mexican Restaurant, 910 Cherokee Rd,...",[{'thumbnail': 'https://lh5.googleusercontent....,['Mexican restaurant'],"{'latitude': 32.935643, 'longitude': -85.952365}",America/Chicago,32.935643,-85.952365
4,La Posada Mexican Grill,https://www.google.com/maps/place/La+Posada+Me...,Mexican restaurant,4.3,http://www.laposadamexicangrill.net/,+1256-329-3005,"La Posada Mexican Grill, 3714 US-280, Alexande...",[{'thumbnail': 'https://lh5.googleusercontent....,"['Mexican restaurant', 'Latin American restaur...","{'latitude': 32.926932, 'longitude': -85.9649187}",America/Chicago,32.926932,-85.964919


The first rows confirm the usual listing shape—restaurant name, a cuisine-style Category, a numeric Rating, and a full U.S.-style Address—and every row in this file now includes a rating, which matches the filter applied before analysis.



In [63]:
df.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 161587 entries, 0 to 161586
Data columns (total 13 columns):
 #   Column           Non-Null Count   Dtype  
---  ------           --------------   -----  
 0   Title            161587 non-null  object 
 1   Link             161587 non-null  object 
 2   Category         161581 non-null  object 
 3   Rating           161587 non-null  float64
 4   Website          133936 non-null  object 
 5   Phone            156396 non-null  object 
 6   Address          161535 non-null  object 
 7   Images           161587 non-null  object 
 8   Categories       161581 non-null  object 
 9   Geo_Coordinates  153361 non-null  object 
 10  Time_Zone        153361 non-null  object 
 11  Latitude         161587 non-null  float64
 12  Longitude        161587 non-null  float64
dtypes: float64(3), object(10)
memory usage: 16.0+ MB


The cleaned table has 161,587 rows (down from ~183.7K after deduplication and again after dropping unrated listings) and 13 columns, still mostly object text; Rating is fully populated (161,587 non-null), while Website, Geo_Coordinates, and Time_Zone still show gaps on a minority of rows.



**Chart rationale (Q3 — bar):**  
I used a **sorted bar chart** of **average rating by cuisine** because Q3 starts as a **category comparison** on a 1–5 scale. Bars make **small differences in means** easy to scan. Color encodes **sample size (`n_restaurants`)** so large groups are not mistaken for more reliable just because the bar is tall. I kept only cuisines with **n ≥ 50** so tiny groups do not create misleading averages.

**What this chart argues:**  
Average ratings **differ by cuisine**, but only **modestly**: in this dataset, group means span roughly **~4.05 to ~4.40** stars—a gap of about **0.3–0.4** on a 5-point scale—with **Chinese** toward the lower end of the averages and **Thai / Vietnamese / Korean / Japanese** slightly higher. The chart argues there is **no dramatic winner/loser cuisine** in stars; differences are **small enough** that they should be read alongside **spread**, not as proof one cuisine is far “better.”


In [64]:
df.describe()


,Rating,Latitude,Longitude
count,161587.000000,161587.000000,161587.000000
mean,4.273971,38.400817,-82.823769
std,0.413718,9.189226,38.835760
min,1.000000,-42.902283,-166.593781
25%,4.100000,34.201443,-103.502839
50%,4.300000,39.680835,-86.125595
75%,4.500000,42.343451,-76.291354
max,5.000000,66.898890,153.365050


Rating summaries are stable and high—mean ~4.27, median 4.3, with most values between about 4.1 and 4.5 on a 1–5 scale—so comparisons of average ratings by cuisine are not skewed by missing stars in this file, though differences between groups remain modest.



In [65]:
df.isnull().sum()


Title                  0
Link                   0
Category               6
Rating                 0
Website            27651
Phone               5191
Address               52
Images                 0
Categories             6
Geo_Coordinates     8226
Time_Zone           8226
Latitude               0
Longitude              0
dtype: int64

Missingness is now concentrated in optional fields: Rating has 0 nulls, only 6 rows lack Category, roughly 17% lack Website, about 5% lack geo/time-zone fields, and a small number lack Address—so Q3 (ratings) can use the full rated subset, while Q2 (state from address) should still account for a few address gaps and Q1 should drop or flag the handful of rows without Category.



---

## Section 3A — Analysis


**Question 1:** Which Asian cuisines (e.g., Chinese, Japanese, Thai, Vietnamese, etc.) are most and least represented among restaurants in the dataset?


In [66]:
# Your analysis for Question 1
# Question 1:
# Which Asian cuisines are most and least represented?

import pandas as pd

# Load cleaned file (created earlier)
df = pd.read_csv("restaurants_cleaned.csv", low_memory=False)

# 1) Keep only rows with a category
work = df.copy()
work["Category"] = work["Category"].fillna("").astype(str).str.strip()
work = work[work["Category"] != ""]

# 2) Map raw category labels -> cuisine groups
# (edit/expand this mapping if you want to include more labels)
cuisine_map = {
    "Chinese": [
        "Chinese restaurant", "Chinese takeaway", "Chinese noodle restaurant",
        "Chinese bakery", "Chinese tea house", "Delivery Chinese restaurant"
    ],
    "Japanese": [
        "Japanese restaurant", "Authentic Japanese restaurant",
        "Japanese steakhouse", "Japanese curry restaurant",
        "Japanese regional restaurant", "Japanese delicatessen",
        "Kyoto style Japanese restaurant"
    ],
    "Thai": ["Thai restaurant"],
    "Vietnamese": ["Vietnamese restaurant"],
    "Korean": [
        "Korean restaurant", "Korean barbecue restaurant", "Korean rib restaurant"
    ],
    "Indian": [
        "Indian restaurant", "North Indian restaurant", "South Indian restaurant",
        "North Eastern Indian restaurant", "Modern Indian restaurant",
        "Indian Muslim restaurant", "Indian takeaway"
    ],
    "Filipino": ["Filipino restaurant"],
    "Malaysian": ["Malaysian restaurant"],
    "Indonesian": ["Indonesian restaurant"],
    "Taiwanese": ["Taiwanese restaurant"],
    # Optional broader buckets:
    "Pan-Asian / General Asian": [
        "Asian restaurant", "Asian fusion restaurant",
        "Pan-Asian restaurant", "South Asian restaurant", "Southeast Asian restaurant"
    ],
}

# Build reverse lookup (raw category -> cuisine group)
reverse_map = {}
for cuisine, labels in cuisine_map.items():
    for label in labels:
        reverse_map[label.lower()] = cuisine

# 3) Assign cuisine group
work["Cuisine_Group"] = work["Category"].str.lower().map(reverse_map)

# Keep only mapped Asian cuisines
asian = work[work["Cuisine_Group"].notna()].copy()

# 4) Count representation
counts = (
    asian["Cuisine_Group"]
    .value_counts()
    .rename_axis("Cuisine")
    .reset_index(name="Restaurant_Count")
)

# Add percent share within Asian subset
counts["Percent_of_Asian"] = (counts["Restaurant_Count"] / counts["Restaurant_Count"].sum() * 100).round(2)

# Most represented
print("Most represented cuisines:")
display(counts.head(5))

# Least represented (non-zero)
print("Least represented cuisines:")
display(counts.tail(5))

# Full table
display(counts)

Most represented cuisines:


,Cuisine,Restaurant_Count,Percent_of_Asian
0,Chinese,10721,40.27
1,Japanese,3582,13.46
2,Thai,3534,13.28
3,Pan-Asian / General Asian,3125,11.74
4,Indian,2181,8.19


Least represented cuisines:


,Cuisine,Restaurant_Count,Percent_of_Asian
6,Korean,1093,4.11
7,Filipino,234,0.88
8,Taiwanese,92,0.35
9,Malaysian,45,0.17
10,Indonesian,34,0.13


,Cuisine,Restaurant_Count,Percent_of_Asian
0,Chinese,10721,40.27
1,Japanese,3582,13.46
2,Thai,3534,13.28
3,Pan-Asian / General Asian,3125,11.74
4,Indian,2181,8.19
5,Vietnamese,1979,7.43
6,Korean,1093,4.11
7,Filipino,234,0.88
8,Taiwanese,92,0.35
9,Malaysian,45,0.17


**Interpretation:**  
Asian cuisine representation in this dataset is uneven. Chinese restaurants are the most represented category by a wide margin, followed by Thai and Japanese, while cuisines such as Indonesian, Malaysian, and Filipino appear much less frequently. This suggests that some Asian cuisines are much more visible on this platform than others, which could reflect differences in real-world restaurant prevalence, platform coverage, or category labeling practices. A key limitation is that cuisine labels are not perfectly standardized (for example, broad labels like “Asian restaurant” and mixed labels like “Asian fusion”), so these counts should be interpreted as approximate representation rather than exact cultural market share.


**Question 2:** How does the representation of Asian cuisines vary across different U.S. states (e.g., which states have higher or lower proportions of Asian cuisine restaurants)?


In [67]:
# Your analysis for Question 2
# Question 2:
# How does representation of Asian cuisines vary across U.S. states?

import pandas as pd

# Load cleaned dataset
df = pd.read_csv("restaurants_cleaned.csv", low_memory=False)

# -----------------------------
# 1) Create an Asian flag
# -----------------------------
pattern = r"\b(?:asian|chinese|japanese|thai|vietnamese|korean|indian|filipino|malaysian|indonesian|taiwanese)\b"

work = df.copy()
work["Category"] = work["Category"].fillna("").astype(str).str.strip()
work["is_asian"] = work["Category"].str.lower().str.contains(pattern, regex=True)

# Exclude false positive explicitly
work.loc[work["Category"].str.lower() == "caucasian restaurant", "is_asian"] = False

# -----------------------------
# 2) Extract U.S. state from Address
# -----------------------------
# Example format: "... , Seattle, WA 98101, United States"
work["Address"] = work["Address"].fillna("").astype(str)
work["State"] = work["Address"].str.extract(
    r",\s*([A-Z]{2})\s+\d{5}(?:-\d{4})?,\s*United States\s*$"
)[0]

# Keep only rows where state was extracted
state_df = work[work["State"].notna()].copy()

# -----------------------------
# 3) State-level representation
# -----------------------------
state_summary = (
    state_df.groupby("State", as_index=False)
    .agg(
        total_restaurants=("State", "size"),
        asian_restaurants=("is_asian", "sum")
    )
)

state_summary["asian_share_pct"] = (
    state_summary["asian_restaurants"] / state_summary["total_restaurants"] * 100
).round(2)

# Optional: require minimum sample size for fair comparison
state_summary_filtered = state_summary[state_summary["total_restaurants"] >= 500].copy()

# Highest and lowest shares
highest = state_summary_filtered.sort_values("asian_share_pct", ascending=False).head(10)
lowest = state_summary_filtered.sort_values("asian_share_pct", ascending=True).head(10)

print("Top 10 states by Asian restaurant share (min 500 total restaurants):")
display(highest)

print("Bottom 10 states by Asian restaurant share (min 500 total restaurants):")
display(lowest)

Top 10 states by Asian restaurant share (min 500 total restaurants):


,State,total_restaurants,asian_restaurants,asian_share_pct
4,CA,17175,4950,28.82
11,HI,1263,332,26.29
3,AZ,2437,631,25.89
10,GA,2388,600,25.13
9,FL,7591,1740,22.92
0,AK,777,161,20.72
47,WA,3275,624,19.05
5,CO,3652,685,18.76
6,CT,3619,620,17.13
14,IL,6232,962,15.44


Bottom 10 states by Asian restaurant share (min 500 total restaurants):


,State,total_restaurants,asian_restaurants,asian_share_pct
41,SD,671,41,6.11
28,ND,822,52,6.33
50,WY,769,54,7.02
32,NM,1396,99,7.09
36,OK,2660,203,7.63
49,WV,1389,106,7.63
26,MT,925,72,7.78
17,KY,1345,105,7.81
23,MN,1600,127,7.94
25,MS,1479,119,8.05


**Interpretation:**  
Representation of Asian cuisines varies substantially across states. Some states have a much higher proportion of Asian cuisine restaurants, while others have relatively low shares, indicating uneven geographic visibility and concentration. This suggests that local demographics, migration patterns, and regional food markets likely influence cuisine representation. A limitation is that state extraction depends on address formatting and category labels, so results are best interpreted as comparative patterns rather than exact market estimates.

**Question 3:** Do average ratings differ across Asian cuisine types, and are certain cuisines consistently rated higher or lower than others?

In [68]:
# Your analysis for Question 3
# Question 3:
# Do average ratings differ across Asian cuisine types?

import pandas as pd

# Load cleaned dataset
df = pd.read_csv("restaurants_cleaned.csv", low_memory=False)

# Keep needed columns and clean
work = df[["Category", "Rating"]].copy()
work["Category"] = work["Category"].fillna("").astype(str).str.strip()
work["rating_num"] = pd.to_numeric(work["Rating"], errors="coerce")

# Keep only valid ratings
work = work[work["rating_num"].notna()]

# Map raw category labels -> cuisine groups
cuisine_map = {
    "Chinese": [
        "Chinese restaurant", "Chinese takeaway", "Chinese noodle restaurant",
        "Chinese bakery", "Chinese tea house", "Delivery Chinese restaurant"
    ],
    "Japanese": [
        "Japanese restaurant", "Authentic Japanese restaurant",
        "Japanese steakhouse", "Japanese curry restaurant",
        "Japanese regional restaurant", "Japanese delicatessen",
        "Kyoto style Japanese restaurant"
    ],
    "Thai": ["Thai restaurant"],
    "Vietnamese": ["Vietnamese restaurant"],
    "Korean": [
        "Korean restaurant", "Korean barbecue restaurant", "Korean rib restaurant"
    ],
    "Indian": [
        "Indian restaurant", "North Indian restaurant", "South Indian restaurant",
        "North Eastern Indian restaurant", "Modern Indian restaurant",
        "Indian Muslim restaurant", "Indian takeaway"
    ],
    "Filipino": ["Filipino restaurant"],
    "Malaysian": ["Malaysian restaurant"],
    "Indonesian": ["Indonesian restaurant"],
    "Taiwanese": ["Taiwanese restaurant"],
    "Pan-Asian / General Asian": [
        "Asian restaurant", "Asian fusion restaurant",
        "Pan-Asian restaurant", "South Asian restaurant", "Southeast Asian restaurant"
    ],
}

# Reverse mapping
reverse_map = {}
for cuisine, labels in cuisine_map.items():
    for label in labels:
        reverse_map[label.lower()] = cuisine

work["Cuisine_Group"] = work["Category"].str.lower().map(reverse_map)

# Keep mapped cuisines only
asian_ratings = work[work["Cuisine_Group"].notna()].copy()

# Aggregate stats by cuisine
q3_summary = (
    asian_ratings.groupby("Cuisine_Group", as_index=False)
    .agg(
        n_restaurants=("rating_num", "size"),
        avg_rating=("rating_num", "mean"),
        median_rating=("rating_num", "median"),
        std_rating=("rating_num", "std")
    )
)

# Keep cuisines with enough observations for stability
q3_summary = q3_summary[q3_summary["n_restaurants"] >= 50].copy()

# Sort by average rating
q3_summary = q3_summary.sort_values("avg_rating", ascending=False)
q3_summary["avg_rating"] = q3_summary["avg_rating"].round(3)
q3_summary["median_rating"] = q3_summary["median_rating"].round(3)
q3_summary["std_rating"] = q3_summary["std_rating"].round(3)

display(q3_summary)

print("Highest average-rated cuisines:")
display(q3_summary.head(5))

print("Lowest average-rated cuisines:")
display(q3_summary.tail(5))

,Cuisine_Group,n_restaurants,avg_rating,median_rating,std_rating
9,Thai,3534,4.396,4.4,0.287
10,Vietnamese,1979,4.375,4.4,0.253
5,Korean,1093,4.365,4.4,0.310
4,Japanese,3582,4.358,4.4,0.305
1,Filipino,234,4.327,4.4,0.508
8,Taiwanese,92,4.312,4.3,0.256
2,Indian,2181,4.294,4.3,0.342
7,Pan-Asian / General Asian,3125,4.272,4.3,0.453
0,Chinese,10721,4.048,4.1,0.388


Highest average-rated cuisines:


,Cuisine_Group,n_restaurants,avg_rating,median_rating,std_rating
9,Thai,3534,4.396,4.4,0.287
10,Vietnamese,1979,4.375,4.4,0.253
5,Korean,1093,4.365,4.4,0.310
4,Japanese,3582,4.358,4.4,0.305
1,Filipino,234,4.327,4.4,0.508


Lowest average-rated cuisines:


,Cuisine_Group,n_restaurants,avg_rating,median_rating,std_rating
1,Filipino,234,4.327,4.4,0.508
8,Taiwanese,92,4.312,4.3,0.256
2,Indian,2181,4.294,4.3,0.342
7,Pan-Asian / General Asian,3125,4.272,4.3,0.453
0,Chinese,10721,4.048,4.1,0.388


**Interpretation:**  
Average ratings do differ across Asian cuisine types, but the differences are relatively modest on a 1–5 scale. Some cuisine groups consistently appear near the top of average ratings, while others are lower, suggesting variation in customer satisfaction (or reviewing behavior) across cuisine categories. To avoid over-interpreting small groups, I applied a minimum sample-size threshold (n >= 50) before comparing means. A limitation is that ratings are platform-generated and may be influenced by location, chain effects, and review volume, so these results show comparative patterns rather than causal quality differences.

---

## Section 3B — Visualization


**Q1:** Which Asian cuisines (e.g., Chinese, Japanese, Thai, Vietnamese, etc.) are most and least represented among restaurants in the dataset?

Chart for Q1

In [70]:
# Your visualization
# Example structure — replace with your actual columns and finding

# fig = px.bar(
#     df,
#     x="your_category_column",
#     y="your_value_column",
#     title="Your finding stated as a claim",
#     labels={"your_category_column": "Readable label", "your_value_column": "Readable label"}
# )
# fig.show()

#most + least represented chart
import textwrap

top5 = counts.nlargest(5, "Restaurant_Count")
bottom5 = counts.nsmallest(5, "Restaurant_Count")

top_bottom = (
    pd.concat([top5, bottom5], ignore_index=True)
      .drop_duplicates(subset="Cuisine")
      .sort_values("Restaurant_Count")
)

_title = (
    "Chinese, Japanese, Thai, general Asian, and Indian categories "
    "dominate category counts"
)
_title_wrapped = "<br>".join(textwrap.wrap(_title, width=62))

fig = px.bar(
    top_bottom,
    x="Restaurant_Count",
    y="Cuisine",
    orientation="h",
    color="Cuisine",
    title=_title_wrapped,
    labels={"Restaurant_Count": "Number of restaurants", "Cuisine": "Cuisine type"},
)
fig.update_layout(margin=dict(t=100))
fig.show()
fig.write_image("most_and_least_represented_asian_cuisines.png")

**What this chart shows:**
Asian cuisine labels in this dataset are highly uneven. Chinese listings dominate mapped Asian categories (~10,700 rows), roughly 3× the next-largest groups (Japanese and Thai, each around 3,500). At the other end, Indonesian, Malaysian, and Taiwanese appear only dozens to low hundreds of times, with Korean and Filipino still far below the leaders. The chart argues that platform category visibility is skewed: a few cuisines take up most of the labeled “Asian restaurant” space, while many others are comparatively rare in the scrape—not necessarily rare in the real world, but underrepresented in how listings are categorized and counted here.

**Chart rationale:**  
I used a “top 5 vs. bottom 5” horizontal bar chart to directly compare the most and least represented Asian cuisine types in the dataset. This format makes the contrast clear without overwhelming the reader with every category. The chart shows that cuisine representation is highly uneven: a few cuisines account for much larger shares, while several cuisines appear relatively rarely. This supports the conclusion that visibility of Asian cuisines on the platform is not evenly distributed.



**Q2:** How does the representation of Asian cuisines vary across different U.S. states (e.g., which states have higher or lower proportions of Asian cuisine restaurants)?

Chart for Q2

In [73]:
#cholopleth of asian restaurants by state
import pandas as pd
import plotly.express as px

# --- Load data ---
df = pd.read_csv("restaurants_cleaned.csv", low_memory=False)

# --- Asian flag (word-boundary; excludes Caucasian false positive) ---
pattern = r"\b(?:asian|chinese|japanese|thai|vietnamese|korean|indian|filipino|malaysian|indonesian|taiwanese)\b"
work = df.copy()
work["Category"] = work["Category"].fillna("").astype(str).str.strip()
work["is_asian"] = work["Category"].str.lower().str.contains(pattern, regex=True)
work.loc[work["Category"].str.lower() == "caucasian restaurant", "is_asian"] = False

# --- US state from address: ", ST 12345, United States" ---
work["Address"] = work["Address"].fillna("").astype(str)
work["State"] = work["Address"].str.extract(
    r",\s*([A-Z]{2})\s+\d{5}(?:-\d{4})?,\s*United States\s*$"
)[0]

us = work[work["State"].notna()].copy()

# --- Aggregate: Asian restaurant count per state ---
choropleth = (
    us.groupby("State", as_index=False)
      .agg(
          total_restaurants=("State", "size"),
          asian_restaurants=("is_asian", "sum"),
      )
)

choropleth["asian_share_pct"] = (
    choropleth["asian_restaurants"] / choropleth["total_restaurants"] * 100
).round(2)

# --- Choropleth: COUNT of Asian restaurants ---
fig = px.choropleth(
    choropleth,
    locations="State",
    locationmode="USA-states",
    color="asian_restaurants",
    scope="usa",
    title="California has the most number of Asian cuisine restaurants by count",
    labels={"asian_restaurants": "Asian restaurants"},
    color_continuous_scale="Blues",
)
fig.update_layout(
    margin=dict(l=0, r=0, t=50, b=0),
    geo=dict(scope="usa", projection_scale=1.05),
)
fig.show()
fig.write_image("asian_cuisine_restaurants_by_state.png")

**What this chart shows:**
California stands out for sheer volume of Asian-flagged listings (~4,950 in this file), with Florida, Illinois, New York, and Texas also ranking high on raw counts. That pattern partly reflects how many total restaurant rows each state has in the export (large markets have more listings overall), not only “more Asian food culture.” The map argues that geographic concentration of Asian-labeled restaurants is real, but count alone can make big states look like the only story.

In [77]:
#cholopleth by percent
fig2 = px.choropleth(
    choropleth,
    locations="State",
    locationmode="USA-states",
    color="asian_share_pct",
    scope="usa",
    title="Asian cuisine's share of each state's total listings tell a different story",
    labels={"asian_share_pct": "Asian share (%)"},
    color_continuous_scale="Reds",
)
fig2.update_layout(margin=dict(l=0, r=0, t=50, b=0), geo=dict(scope="usa"))
fig2.show()
fig2.write_image("share_of_restaurants_that_are_asian_cuisine.png")

**What this chart shows:**
When each state is measured by share (Asian-flagged rows ÷ all rows in that state), California still ranks high (~29%), but states like Hawaii and Arizona also show strong proportional presence even if their absolute counts are smaller. The second map argues that “many Asian restaurants” and “Asian cuisine is a large fraction of that state’s listings” are not the same claim—readers need both views to avoid confusing market size with relative visibility in the data.

**What these maps show together:**
The count map highlights where the database has the most Asian-labeled listings (led by California). The share map highlights where Asian labels make up a larger slice of that state’s rows—including states that are prominent on percent but not always #1 on raw count. Together they argue that state-level “Asian food presence” in this dataset depends on which metric you use, and neither map alone is sufficient.

**Chart rationale:**  
I used two choropleth maps side by side because they answer different but related questions about the same geography. The first map plots the number of Asian-cuisine restaurants per state; it highlights where the largest totals appear (often states with bigger populations and more listings overall). The second map plots share (% of restaurants in each state flagged as Asian cuisine); it adjusts for scale and makes it easier to compare whether Asian cuisine listings are unusually common or uncommon in smaller states versus larger ones.

Together, these views reduce misinterpretation: a state can rank high on counts but only moderate on share, meaning “many listings” partly reflects marketplace size—not only cultural food presence. Readers should interpret both maps relative to limitations in the dataset (category labels as proxies for cuisine and address-based state parsing dropping some rows). The takeaway should combine both signals: geographic concentration and proportional representation within each state’s slice of the data.

**Q3:** Do average ratings differ across Asian cuisine types, and are certain cuisines consistently rated higher or lower than others?

Chart for Q3


In [78]:
import plotly.express as px

fig = px.bar(
    q3_summary.sort_values("avg_rating", ascending=False),
    x="Cuisine_Group",
    y="avg_rating",
    color="n_restaurants",
    title="Mean ratings by cuisine are very similar",
    labels={"avg_rating": "Average rating", "Cuisine_Group": "Cuisine type", "n_restaurants": "Sample size"}
)
fig.update_layout(xaxis_tickangle=-35)
fig.show()

**What this chart shows:**
Average ratings differ by cuisine, but only modestly on a 1–5 scale: in this run, group means span roughly 4.05 (Chinese) to ~4.40 (Thai / Vietnamese / Korean / Japanese), a gap of about 0.35 stars, with color hinting at sample size (Chinese has the largest n). The chart argues there is no dramatic “winner/loser” cuisine in stars—differences are small enough that they should be read alongside spread and review behavior, not as proof one cuisine is “better.”

**Chart Rationale:**
I used a sorted bar chart of average rating by cuisine group because Question 3 is fundamentally a comparison across categories: readers need to see which cuisines sit slightly higher or lower on a shared 1–5 scale. Bars make those small differences in means easy to scan, especially when cuisines are ordered from highest to lowest average. I colored bars by sample size (n_restaurants) so large groups (for example Chinese) are not mistaken for more reliable just because the bar is prominent—readers can see both level and how much data supports each mean. I applied a minimum n ≥ 50 rule before plotting so tiny groups do not create misleading spikes. I chose this chart instead of only the jittered scatter because means are a clear summary for “are averages different?”—while the scatter (below) is better for showing overlap in the full distributions. The main takeaway this chart is meant to support: average ratings differ modestly across cuisines, not dramatically—so any ranking should be read as a slight shift, not proof that one cuisine is far “better” than another in this listing data.

In [79]:
import numpy as np
import pandas as pd
import plotly.express as px
import textwrap

rng = np.random.default_rng(42)
# Per-restaurant rows with mapped cuisine (defined in the Question 3 prep cell)
plot = asian_ratings.copy()
cat = pd.Categorical(plot["Cuisine_Group"])
cuisine_order = list(cat.categories)
plot["xj"] = cat.codes.astype(float) + rng.uniform(-0.25, 0.25, size=len(plot))

_title = (
    "Restaurant-level ratings overlap heavily across cuisines "
    "despite small differences in averages"
)
_title_wrapped = "<br>".join(textwrap.wrap(_title, width=62))

fig = px.scatter(
    plot,
    x="xj",
    y="rating_num",
    color="Cuisine_Group",
    labels={"rating_num": "Rating (1–5)", "xj": "Cuisine type (jittered)"},
    title=_title_wrapped,
)

fig.update_traces(marker=dict(size=6, opacity=0.25))
fig.update_layout(
    height=700,
    showlegend=True,
    xaxis=dict(
        title="Cuisine type (horizontal jitter for visibility)",
        tickmode="array",
        tickvals=list(range(len(cuisine_order))),
        ticktext=cuisine_order,
        tickangle=-35,
    ),
)
fig.show()

**What this chart shows:**
At the individual restaurant level, ratings for most cuisines pile up between about 4.0 and 4.5, with heavy overlap between groups—many cuisines share the same cloud, not separate bands. The scatter argues that means in the bar chart understate how similar distributions are: a small average gap can still sit on top of mostly shared rating habits (lots of 4.3–4.4 listings everywhere). It supports reading Q3 as “slight shifts and overlap,” not “one cuisine is clearly rated far above the rest.”

**Chart Rationale**
I used a jittered strip scatter plot instead of only plotting averages because ratings rarely behave like a single number—within each cuisine type many restaurants cluster near typical ratings while others fall higher or lower. Spreading points horizontally with jitter separates overlapping markers so density is visible rather than mashed into one vertical stack. Transparency keeps large groups readable without implying false precision about individual listings. Together, this chart supports the Question 3 claim that ratings differ between cuisines and clarifies whether differences look like a consistent shift upward/downward versus wide overlap in the distributions.



---

## Section 4 — Conclusions


This analysis used a deduplicated restaurant dataset to compare how Asian cuisines appear in platform category labels across the United States and how ratings differ by mapped cuisine groups. Cuisine representation is uneven: Chinese, Thai, and Japanese categories appear among the largest groups, while several other Asian cuisines occur much less often. Geography matters in two ways: some states show many Asian-cuisine listings in absolute terms, but state-level shares (percent of listings) tell a different story about proportional visibility. Ratings differ modestly across cuisine groups, though distributions overlap—so averages should be read cautiously alongside spread and sample size.

**Q1 — Which cuisines are most and least represented?**
Chinese listings dominate mapped Asian categories in this dataset—far more than Japanese, Thai, or Indian—while cuisines such as Indonesian, Malaysian, and Taiwanese appear only a few dozen to a few hundred times. That suggests uneven visibility in how platforms label Asian food, not necessarily that those cuisines are equally rare in real life. I would next refine the cuisine taxonomy (e.g. split “Asian restaurant” / fusion labels) and compare counts to an external reference (census or industry data) to see how much is labeling vs reality.

**Q2 — How does representation vary by state?**
California has the largest raw count of Asian-flagged listings, but share (percent of each state’s rows) tells a different story—states like Hawaii and Arizona also show strong proportional presence. That suggests you should not equate “big state on the map” with “Asian cuisine is most common there” without looking at both metrics. With more time I would map metros instead of only states, fix address parsing gaps, and filter small states more consistently when comparing shares.

**Q3 — Do ratings differ by cuisine type?**
Average ratings differ modestly across cuisine groups (roughly a third of a star between the lowest and highest means), and restaurant-level ratings overlap heavily—most cuisines cluster around the same 4.0–4.5 band. That suggests small average gaps are not strong evidence that one cuisine is “better” in this scrape; reviewer behavior, chains, and sample size matter. I would add review counts, price tier, and city-level controls, and test whether differences hold after filtering chains or low-review listings.






**What is the most important thing your analysis revealed?**
The dataset does not reflect Asian cuisines evenly: a few cuisines dominate category counts while others are relatively rare. That imbalance affects how visible different cultures appear on this kind of listings platform—not only where restaurants exist.

**What surprised you?**
(Pick what matched your plots.) I was struck by how large the gap can be between the most and least represented cuisines after cleaning duplicates, and by how much counts vs. state share can disagree when comparing states.

**What would you investigate next if you had more time or data?**
I would add review count and price level (if available), compare metro areas instead of only states, and improve location parsing with coordinates. I would also test a stricter cuisine taxonomy (e.g., using Categories or external labels) to reduce reliance on a single Category string.

**What are the limitations — what can't you conclude?**
You cannot claim true population demographics or causal “food quality” from star ratings alone. Categories are platform-defined and inconsistent; Pan-Asian / Restaurant buckets blur cuisine identity; Address-based states omit some rows; deduping rules (Title+Address+Phone+Lat+Lon) change totals; ratings can reflect chains, reviewer bias, listing completeness, and missing ratings. At best this analysis shows patterns in scraped/listing labels and aggregates, not a complete census of U.S. dining.

---
## Section 5 — My Process

**Starting point** 
I was mainly curious about these 3 questions: which Asian cuisines show up most in listing data, how that varies by U.S. state, and whether star ratings differ by cuisine type. I used the Kaggle 380K U.S. restaurant export because it fit that scope and my HCD interest in how platforms label and surface cultural identity—not because it is a full census of every restaurant in the country.

**First pass; profiling the raw file**
I started with 380K_US_Restaurants.csv and ran head, info, describe, and isnull. The table looked usable at a glance, but profiling showed many duplicate rows (same title + address repeated) and messy text (Category as hundreds of strings, Rating as text, missing geo/contact fields). I flagged Asian-related rows with regex and excluded false positives like “Caucasian restaurant.” That pushed me to treat raw row count as listing rows, not unique restaurants, until I cleaned the file.

**First cleaning: deduplication**
The first cleaning pass focused on deduplication. I grouped duplicate clusters (same Title and Address, checking phone and coordinates) and removed repeats so one venue was not counted many times. That cut the file from about 380,358 rows to ~183,727 and produced restaurants_cleaned.csv. Without this pass, Q1 counts and Q2 state totals would have been inflated.


**Second cleaning: rated listings only**
I profiled restaurants_cleaned.csv again and saw that Q3 still had missing or invalid ratings on roughly 12% of rows. The second cleaning pass coerced Rating to numeric, dropped rows without a valid rating, and overwrote restaurants_cleaned.csv. That left about 161,587 rows with no missing ratings, so Q3 charts compare cuisines on the same rule. The notebook documents both counts so it is clear the file was cleaned twice for two reasons—duplicates first, then rating completeness.

**Analysis and visualization (after cleaning)**
Then I mapped Category → Cuisine_Group, aggregated for Q1, parsed state and computed count and share (%) for Q2, and used means plus a jittered scatter for Q3. Using two choropleths for Q2 changed how I read “where Asian cuisine looks common.” Pairing the mean bar with the scatter kept me from over-reading small average gaps when distributions overlapped.



**Summary of findings:**  
*This analysis used a deduplicated U.S. restaurant listing dataset to see how Asian cuisines show up in platform category labels and how star ratings vary across mapped cuisine groups. Representation is uneven: Chinese, Thai, and Japanese labels sit among the largest buckets, while several other Asian cuisines appear far less often—so visibility in the data is not culturally even. Geography matters in two ways at once: a few states stand out for raw counts of Asian-cuisine listings, but state-level share (percent of that state’s listings) paints a different picture of proportional presence. Ratings differ only modestly across cuisines, and the distributions overlap heavily, which means small gaps in averages should be read alongside spread and sample size rather than as strong rank-ordering of “quality.” Overall, these patterns describe listing labels and aggregates on a platform, not a full census of dining or demographics, and they should not be stretched into claims about population mix or causal quality.*

---

## Competency Claim

In a `mp1.md` file in your GitHub repository, write a short competency claim (2–4 sentences) for each domain you feel this project demonstrates. Be specific — cite something you actually did in this notebook.

Domains covered by this project typically include:
- **C3 — Data cleaning and file handling** (if you cleaned or reshaped data)
- **C5 — Data analysis with pandas** (answering questions with code)
- **C6 — Data visualization** (your chart)
- **C7 — Critical evaluation and professional judgment** (your interpretation and limitations section)

You don't have to claim every domain — only the ones your work actually demonstrates.